modules/raft/src/main/java/org/apache/ignite/internal/raft/storage/segstore/IndexMemTable.java

```diff
@@ -64,6 +64,22 @@ public SegmentInfo segmentInfo(long groupId) {
         return stripe(groupId).memTable.get(groupId);
     }
 
+    @Override
+    public void truncateSuffix(long groupId, long lastLogIndexKept) {
+        ConcurrentMap<Long, SegmentInfo> memtable = stripe(groupId).memTable;
+
+        SegmentInfo segmentInfo = memtable.get(groupId);
+
+        if (segmentInfo == null || lastLogIndexKept < segmentInfo.firstLogIndexInclusive()) {
+            // If the current memtable does not have information for the given group or if we are truncating everything currently present
+            // in the memtable, we need to write a special "empty" SegmentInfo into the memtable to override existing persisted data during
+            // search.
+            memtable.put(groupId, new SegmentInfo(lastLogIndexKept + 1));
```

Comment:
I suppose there's another implicit reason to do so: `SegmentInfo#logIndexBase` is final, the "head" of the array is immutable. If you change it you must create a new array. Worth mentioning I suppose

Reply:
Otherwise it's hard for me personally to justify the "need" to replace the old array with a new one, maybe I'm missing something

Reply:
Ok, I see.
> I wanted to emphasize in this comment that this SegmentInfo is special

My point is - why does it matter? Why can't existing method process the special case? You may leave this comment intact, but in my opinion it does not explain enough